python/ray/dag/compiled_dag_node.py

```diff
@@ -101,8 +111,11 @@ def do_exec_compiled_task(
                 self._output_writer.write(wrapped)
             else:
                 self._output_writer.write(output_val)
-            finally:
+
+            try:
                 self._input_reader.end_read()
+            except BlockingIOError:
+                pass
```

Comment:
Yes, that's correct. Though I just changed the pass to a break to make this clearer.

I took out the finally block because I wanted to wrap the call to end_read() in a try/except block without needing the extra indentation imposed by the finally block. Do you want me to add it back in? It doesn't appear that the finally block is necessarily providing additional functionality because there is no return statement in the try or except blocks.

Reply:
No, that's all right, thanks for the explanation!